WordPress/gutenberg
# Add template areas to template details

## Description

Address a part of https://github.com/WordPress/gutenberg/issues/29147.

Add **Template Areas** section to the template details in the header dropdown. This PR focuses on adding the functionality, styling will be done in other PRs.

## How has this been tested?



1. Activate `tt1-block` theme
2. Go to Site Editor
3. View a template and click to open the dropdown in the top bar
4. Should see the template areas of the template
5. Hover/Focus on template areas should highlight the corresponding blocks
6. Click on them should select the corresponding blocks

## Screenshots 
![image](https://user-images.githubusercontent.com/7753001/135192873-997c279d-ab76-4727-96e7-d6fcaf115b95.png)

## Types of changes




New feature

## Checklist:
